### DISCUSSION TITLE
Syntax Highlighting Configuration

### DISCUSSION THREAD

**Discussion by jmartinWestern** [nix-community/nixvim#2458] [Q&A]
How can I disable the syntax highlighting provided by the nixd lsp server and use the syntax highlighting provided by the vim-nix plugin?

My nixvim config is on top, my previous neovim config is on bottom.

![Screenshot 2024-10-25 at 10 33 36 AM](https://github.com/user-attachments/assets/427596ed-f357-40cc-8792-fc331e642b48)

**Comment by MattSturgeon** [comment]
I'm out and about currently, so can't investigate a "proper" answer.

I think it's nixd's _Semantic Tokens_ feature, which you should be able to disable if you prefer.

https://github.com/nix-community/nixd/blob/main/nixd%2Fdocs%2Ffeatures.md

**Comment by jmartinWestern** [comment]
This pointed me in the right direction.  I appreciate your help!

**Reply by MattSturgeon** [reply]
Awesome! I'm always interested in seeing the final solution if you're able to share :grin:

**Reply by jmartinWestern** [reply]
I found this code block at https://www.reddit.com/r/neovim/comments/zjqquc/how_do_i_turn_off_semantic_tokens/ 

```lua
vim.api.nvim_create_autocmd("LspAttach", {
  callback = function(args)
    local client = vim.lsp.get_client_by_id(args.data.client_id)
    client.server_capabilities.semanticTokensProvider = nil
  end,
});
```

I added it as extraConfigLua and it accomplished what I needed.  It does disable semantic tokens for all LSPs, but that hasn't affected my use case.

**Reply by MattSturgeon** [reply]
Awesome!

You can leverage some of nixvim's options to do that in a more nix-module way if you prefer:

```nix
plugins.lsp.onAttach = ''
  client.server_capabilities.semanticTokensProvider = nil
'';
```

Or
```nix
plugins.lsp.capabilities = ''
  capabilities.semanticTokensProvider = nil
'';
```

If you would rather do it specifically for nixd, however, you could use the server-specific options:

- [`plugins.lsp.servers.nixd.onAttach`](https://nix-community.github.io/nixvim/plugins/lsp/servers/nixd/onAttach.html#pluginslspserversnixdonattach) overrides or extend the "default" or "global" onAttach specifically for nixd
- [`plugins.lsp.servers.nixd.extraOptions`](https://nix-community.github.io/nixvim/plugins/lsp/servers/nixd/index.html#pluginslspserversnixdextraoptions) and [`plugins.lsp.servers.nixd.settings`](https://nix-community.github.io/nixvim/plugins/lsp/servers/nixd/settings/index.html#pluginslspserversnixdsettings)
  can be used to pass extra configuration to nixd
  
```nix
plugins.lsp.servers.nixd.onAttach = {
  # Only run when starting nixd
  function = ''
    client.server_capabilities.semanticTokensProvider = nil
  '';

  # Set to true if you don't want to run the default `plugins.lsp.onAttach` too
  override = false;
}
```

**Reply by jmartinWestern** [reply]
That's fantastic.  I knew there was a way to do it with the nix-module, but I couldn't quite work it out.
